src/clocksi_downstream.erl

```diff
@@ -35,8 +35,8 @@
 %% @doc Returns downstream operation for upstream operation
 %%      input: Update - upstream operation
 %%      output: Downstream operation or {error, Reason}
--spec generate_downstream_op(Transaction :: tx(), Node :: index_node(), Key :: key(),
-  Type :: type(), Update :: op_param(), list()) ->
+-spec generate_downstream_op(tx(), index_node(), {key(), bucket()},
+    type(), op_param(), list()) ->
```

Comment:
Makes it clear that key here is always a tuple of key and bucket -> relevant for assumptions in bcounter_mgr